### ISSUE TITLE
[Feature] deepseek_v4_flash_dspark: emit the drafter input (dspark_target_hidden) from the target forwards

### ISSUE THREAD

**Issue by ndleslx** [hw-native-sys/pypto-lib#1078]
### Summary

`models/deepseek_v4_flash_dspark` target programs cannot feed the DSpark drafter
they were built for: `l3_decode_fwd` (`decode_fwd.py:1029`) and `l3_prefill_fwd`
(`prefill_fwd.py:1001`) return only `(x_out, logits, sampled_ids)`, while
`l3_dspark_drafter` (`dspark_drafter.py:725`) consumes

```python
target_hidden: pl.Tensor[[N_RANKS, T_MAIN_DYN, MAIN_IN], pl.BF16]   # MAIN_IN = 3 * D
```

(`dspark_drafter.py:726`), where row `t` concatenates the `hc_head`-projected
hidden states of target layers 40/41/42 — `dspark_proj.py:27-28`
(`TARGET_LAYERS = 3  # dspark_target_layer_ids`, `MAIN_HIDDEN_DIM = 3 * D`),
matching `dspark_target_layer_ids: [40, 41, 42]` in the release config. Nothing
in the directory produces this tensor; every drafter harness (`dspark_drafter.py`
`__main__`, `dspark_prefill.py`) feeds it a synthetic fixture. Request: add a
`dspark_target_hidden` output to both target host drivers, with fixture/golden
updates.

### Motivation / Use Case

This is the missing piece of #905 P6 ("DSpark drafter — operators landed (#947);
wiring open"). DSpark serving chains the drafter **after** the target verify step:
the drafter's context-KV insertion for step N+1 projects exactly the
layer-40/41/42 stacked hidden of the rows the target verified in step N
(`prepare_dspark_inputs`, `dspark_drafter.py:212`). pypto-serving is integrating
this directory for end-to-end generation; without the output tap, a serving
runtime would have to re-dispatch the last three target layers every decode step
— extra dispatches and weight-bank re-staging that defeat the fused 43-layer
program.

### Proposed API / Behavior

```python
# decode_fwd.py — decode_fwd / l3_decode_fwd
dspark_target_hidden: pl.Out[pl.Tensor[[N_RANKS, T_DYN, 3 * D], pl.BF16]]
# prefill_fwd.py — prefill_fwd / l3_prefill_fwd (CP-local prompt rows; the drafter's
# own dspark_cp_allgather_hidden reconstructs the group rows)
dspark_target_hidden: pl.Out[pl.Tensor[[N_RANKS, FWD_TOKENS_DYN, 3 * D], pl.BF16]]
```

- Tap points: apply `hc_head` to the post-layer hc stack of layers 40 and 41
  inside the unrolled layer loop — the same projection already applied to the
  final stack at `decode_fwd.py:1007` / `prefill_fwd.py:985` — and reuse the
  existing `hidden_workspace` for layer 42.
- Keep `TARGET_LAYERS` / `MAIN_HIDDEN_DIM` (`dspark_proj.py:27-28`) the single
  source of the tap layer ids so the forwards and the drafter cannot drift.
- Row order is the target's own row order (decode: anchor + 7 verify rows per
  request; prefill: CP-local prompt rows) — what `build_tensor_specs(mode=...)`
  in `dspark_drafter.py:948` already assumes.
- Fixtures: extend `build_tensor_specs` in both entries and compare via the same
  `compare_fn` route used for `x_out` / `sampled_ids` (the decode golden oracle
  was dropped in #1062).

### Alternatives Considered

- Serving re-dispatches the last three target layers per step: extra L3
  dispatches plus weight-bank re-staging; rejected.
- The drafter recomputes the projection from a copied full hc stack: still needs
  the three per-layer stacks exported — same ABI surface, more data movement.
- Keep the synthetic-fixture chaining harness-only: blocks any real serving
  integration (#905 P6 stays open).

### Additional Context

Related: #905 (P6 — drafter wiring), #1004 (P5 — the prefill tap must respect
the DSA-CP row layout).
